artiq/frontend/artiq_dashboard.py

```diff
@@ -40,6 +39,12 @@ def get_argparser():
     parser.add_argument(
         "--port-broadcast", default=1067, type=int,
         help="TCP port to connect to for broadcasts")
+    parser.add_argument(
+        "--port-proxy-core-pubsub", default=1383, type=int,
+        help="TCP port to connect to for core device proxy (for data)")
+    parser.add_argument(
+        "--port-proxy-core-rpc", default=1384, type=int,
+        help="TCP port to connect to for core device proxy (for control)")
```

Comment:
Just read the device database (obtained from the master) and figure out the proxy address from its controller entry.

Reply:
And of course, disable moninj cleanly and do not crash if such an entry cannot be found.